src/app/Mappings/components/MappingsTable.tsx

```diff
@@ -1,13 +1,119 @@
+import VMwareProviderHostsTable from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderHostsTable';
+import MappingsActionsDropdown from './MappingsActionsDropdown';
 
 interface IMappingsTableProps {
   mappings: Mapping[];
+  mappingType: string;
+  toggleAddEditModal: () => void;
 }
 
 const MappingsTable: React.FunctionComponent<IMappingsTableProps> = ({
   mappings,
+  mappingType,
+  toggleAddEditModal,
 }: IMappingsTableProps) => {
+  const getSortValues = (mapping: ICommonMapping) => {
+    const { name, sourceProvider, targetProvider } = mapping;
+    return [name, sourceProvider.name, targetProvider.name, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(mappings, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const {
+    selectedItems: expandedMappings,
+    toggleItemSelected: toggleMappingExpanded,
+  } = useSelectionState<ICommonMapping>(sortedItems);
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable] },
+    { title: 'Source provider', transforms: [sortable] },
+    { title: 'Target provider', transforms: [sortable] },
+    {
+      title: <>{mappingType === MappingType.Network ? 'Network mappings' : 'Storage mappings'}</>,
+      transforms: [sortable],
+      cellTransforms: [compoundExpand],
+    },
+    { title: '', columnTransforms: [classNamesTransform(tableStyles.tableAction)] },
+  ];
+
+  const rows: IRow[] = [];
+  currentPageItems.forEach((mapping: ICommonMapping) => {
+    const { name, sourceProvider, targetProvider, items } = mapping;
+    const isExpanded = expandedMappings.includes(mapping);
+    rows.push({
+      meta: { mapping },
+      isOpen: isExpanded,
+      cells: [
+        name,
+        sourceProvider.name,
+        targetProvider.name,
+        {
+          title: (
+            <>
+              {mappingType === MappingType.Network ? (
+                <NetworkIcon key="hosts-icon" />
+              ) : (
+                <DatabaseIcon key="storage-icon" />
+              )}{' '}
+              {items ? items.length : 0}
+            </>
+          ),
+          props: {
+            isOpen: isExpanded,
+          },
+        },
+        {
+          title: <MappingsActionsDropdown />,
+        },
+      ],
+    });
+    if (isExpanded) {
+      rows.push({
+        parent: rows.length - 1,
+        compoundExpand: 4,
+        cells: [
+          {
+            title: (
+              <div>
+                TODO: mapping details table
+                {/* <MappingDetailsTable {...props} /> */}
+              </div>
+            ),
+            props: { colSpan: columns.length, className: tableStyles.modifiers.noPadding },
+          },
+        ],
+      });
+    }
+  });
   // TODO: the storage and network mappings tables seem similar enough that we
```

Comment:
We can get rid of my TODO comment here I think, now that we have some code for this.